netbox/extras/models/customfields.py

```diff
@@ -281,36 +283,48 @@ def populate_initial_data(self, content_types):
         Populate initial custom field data upon either a) the creation of a new CustomField, or
         b) the assignment of an existing CustomField to new object types.
         """
+        if self.default is None:
+            # We have to convert None to a JSON null for jsonb_set()
+            value = RawSQL("'null'::jsonb", [])
+        else:
+            value = Value(self.default, models.JSONField())
         for ct in content_types:
-            model = ct.model_class()
-            instances = model.objects.exclude(**{'custom_field_data__contains': self.name})
-            for instance in instances:
-                instance.custom_field_data[self.name] = self.default
-            model.objects.bulk_update(instances, ['custom_field_data'], batch_size=100)
+            ct.model_class().objects.update(
+                custom_field_data=Func(
+                    F('custom_field_data'),
+                    Value([self.name]),
+                    value,
+                    function='jsonb_set'
+                )
+            )
 
     def remove_stale_data(self, content_types):
         """
         Delete custom field data which is no longer relevant (either because the CustomField is
         no longer assigned to a model, or because it has been deleted).
         """
         for ct in content_types:
-            if model := ct.model_class():
-                instances = model.objects.filter(custom_field_data__has_key=self.name)
-                for instance in instances:
-                    del instance.custom_field_data[self.name]
-                model.objects.bulk_update(instances, ['custom_field_data'], batch_size=100)
+            ct.model_class().objects.update(
+                custom_field_data=F('custom_field_data') - self.name
+            )
```

Comment:
Previously there was a check here (but oddly not in the other modified methods) to make sure the result of `ct.model_class()` wasn't falsy (I assume `None`, which I've seen locally once or twice).

Is the feeling that it's safe to assume that we'll never try to call `None.objects`?

Reply:
Good catch. I suspect it might be there to elegantly handle the case where a plugin model has been removed. I'm going to restore the check.